index.js

```diff
@@ -1,9 +1,9 @@
-module.exports = function reverse (a) {
-  var length = a.length
-  var buffer = new Buffer(length)
+module.exports = function reverse (src) {
+  var buffer = new Buffer(src.length)
 
-  for (var i = 0, j = length - 1; i < length; ++i, --j) {
-    buffer[i] = a[j]
+  for (var i = 0, j = src.length - 1; i <= j; ++i, --j) {
+    buffer[i] = src[j]
+    buffer[j] = src[i]
```

Comment:
it just occured to me that by using a temporary variable you could do an in-place reverse, which would skip the allocation. the only module that depends on this is bitcoinjs-lib, which needs to reverse hashes (so, mutable isn't a problem). However, that library seems to just use `[].reverse.call(buffer)` although it does depend on this module.